### PR TITLE
Add family name in the guardio insurance sample application

### PR DESCRIPTION
**Pull request by Achintha444**

### CODE REVIEW

**Review by madurangasiriwardena — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ export default function decodeMeObject(user) {
+
+    if(user.id==undefined || user.userName==undefined || user.name==undefined || user.emails==undefined ) {
```
Comment (sandushi) [inline b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js:21]: formatting issue

File: b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js [hunk 2]
```diff
@@ -33,9 +33,16 @@ export default function EditUserComponent(props) {
 
     const [loadingDisplay, setLoadingDisplay] = useState(LOADING_DISPLAY_NONE);
 
-    const nameValidate = (name, errors) => {
-        if (!name) {
-            errors.name = 'This field cannot be empty'
+    const firstNameValidate = (firstName, errors) => {
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js:36]: These validators can be generalized to a single validator.

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js [hunk 3]
```diff
@@ -75,12 +82,13 @@ export default function AddUserComponent(props) {
...
         const errors = {}
-        errors = nameValidate(values.name, errors);
+        errors = firstNameValidate(values.firstName, errors);
+        errors = familyNameValidate(values.familyName, errors);
         errors = emailValidate(values.email, errors);
         errors = usernameValidate(values.username, errors)
         errors = passwordValidate(values.password, errors);
         errors = repasswordValidate(values.repassword, errors);
-
+        console.log(errors);
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js:91]: Shall we remove console logs?

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js [hunk 4]
```diff
@@ -25,7 +25,8 @@ export default function decodeUser(user) {
...
         "id": user.id,
         "username": user.userName,
-        "name": user.name != undefined ? user.name.givenName : "Not Defined",
-        "email": user.emails != undefined ? user.emails[0] : "Not Defined"
+        "firstName": user.name != undefined ? user.name.givenName : "-",
+        "familyName": user.name != undefined ? user.name.familyName : "-",
+        "email": user.emails != undefined ? user.emails[0] : "-"
     };
-}
+}
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js:32]: Let's add a new line at the end of the file.

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js (hunk 3 above)
Comment (lashinijay) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js:91]: Check other places too.

File: b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js (hunk 2 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/editUser/editUserComponent.js:36]: This is fixed with a later PR,

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/libs/shared/util/util-front-end-util/src/lib/formUtil/formUtil.ts 
- https://github.com/wso2/samples-is/blob/09d8f7d6abe21d199f2c739e578b0883825d8cf2/b2b-sample/apps/business-admin-app/components/sections/sections/settingsSection/manageUserSection/otherComponents/addUserComponent.tsx#L60

### Commit
- https://github.com/wso2/samples-is/commit/9c330e205151b20f9436bf6aad9c4db85e216673#diff-2982d16cf307c73b8cbce49859f3040b0fa42f10c49acaa3952642cdd40632b5

File: b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js (hunk 3 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/settingsComponents/addUserComponent.js:91]: These have been removed with later PR's, as there are no console.log 's in unnecessary places.

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js (hunk 4 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeUser.js:32]: Fixed with https://github.com/Achintha444/samples-is/commit/4120b50f44ba393f00f7c7cf57aa04d21f6bc8d6#diff-63d09dc68fe77aadebf84c960e9e5bd8b971f4e692ac00404169d3487ae387a5

File: b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js (hunk 1 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/util/util/apiUtil/decodeMeObject.js:21]: This is fixed with later PR's

### File
- https://github.com/wso2/samples-is/blob/master/b2b-sample/libs/business-app/data-access/data-access-controller/src/lib/controller/controllerMe/controllerDecodeMe.ts

### Commit
- https://github.com/wso2/samples-is/commit/c56439215193067034ac953d02cb38272b2be8d4#diff-41837f5472f12624dbae0a8457f15e50adb73bf6a7f5f2aac6228a4616a1d56f

**Pull request merged**